### PR TITLE
Add Scala 3.0.1 cross-build

### PR DESCRIPTION
**Pull request by sergiy0**
Could you also be so kind to create a new release with these changes upon approval and merge of this PR?

### CODE REVIEW

File: kubernetes-client/resources/swagger/k8s-1.18-swagger.json [hunk 1]
```diff
@@ -15771,7 +15771,7 @@
         "description": {
           "type": "string"
         },
-        "enum": {
+        "`enum`": {
```
Comment (armanbilge) [inline kubernetes-client/resources/swagger/k8s-1.18-swagger.json]: You shouldn't be changing these in the swagger.json, instead you should fix it here:
https://github.com/joan38/kubernetes-client/blob/master/project/SwaggerModelGenerator.sc#L148-L154

File: kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala [hunk 2]
```diff
@@ -50,8 +50,14 @@ private[client] trait Creatable[F[_], Resource <: { def metadata: Option[ObjectM
...
       .flatMap {
         case status if status.isSuccess => update
-        case Status.NotFound            => create(resource).recoverWith { case Status.Conflict => update }
-        case status                     => F.pure(status)
+        case Status.NotFound =>
+          create(resource)
+            .flatMap {
+              case Status.Conflict => F.raiseError[Status](UnexpectedStatus(Status.Conflict, GET, fullResourceUri))
+              case status          => F.pure(status)
+            }
```
Comment (armanbilge) [inline kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala]: Why did this change?

**Review by armanbilge — CHANGES_REQUESTED** [review]

File: kubernetes-client/resources/swagger/k8s-1.18-swagger.json (hunk 1 above)
Comment (joan38) [reply kubernetes-client/resources/swagger/k8s-1.18-swagger.json]: Fixed

File: kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala (hunk 2 above)
Comment (joan38) [reply kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala]: The previous code was not correct and Scala 3 makes it not compile rightfully.

Comment (armanbilge) [reply kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala]: Thanks for the explanation 👍 my question should have been "why did this change so extensively", the updated diff here looks more like what I would expect :)

**Comment by joan38** [comment]
Thanks you both @armanbilge and @sergiy0

**Comment by joan38** [comment]
`0.8.0` released with Scala 3 support 🥳

**Pull request merged**